src/pyflask/manageNeuroconv/manage_neuroconv.py

```
@@ -674,25 +680,116 @@ def validate_metadata(metadata: dict, check_function_name: str) -> dict:
+            value = info["values"].get(method, None)
+            if value != None:
+
+                try:
+                    if method == "timestamps":
+
+                        # Open the input CSV file for reading
+                        with open(value, mode="r", newline="") as csvfile:
+                            reader = csv.reader(csvfile)
+                            rows = list(reader)
+                            timestamps_array = [float(row[0]) for row in rows]
+
+                            # NOTE: Not sure if it's acceptable to provide timestamps of an arbitrary size
+                            # Use the provided timestamps to align the interface
+                            if hasattr(interface, "sorting_extractor"):
+                                if not interface.sorting_extractor.has_recording():
+                                    extractor = interface.sorting_extractor
+                                    fs = extractor.get_sampling_frequency()
+                                    end_frame = len(timestamps_array)
+                                    mock_recording_interface = MockRecordingInterface(
+                                        sampling_frequency=fs, durations=[end_frame / fs], num_channels=1
+                                    )
+                                    interface.register_recording(mock_recording_interface)
+
+                            interface.set_aligned_timestamps(np.array(timestamps_array))
+
+                    # Register the linked interface
+                    elif method == "linked":
+                        interface.register_recording(converter.data_interface_objects[value])
+
+                    elif method == "start":
+                        interface.set_aligned_starting_time(
+                            value
+                        )  # For sorting interfaces, an empty array will still be returned
+
+                except Exception as e:
+                    errors[name] = str(e)
```

I'm almost done refactoring a bit here and there on the backend for this

But how did you want the frontend to handle this error step given the recent advances in logging?

---

I.e., can we just let them raise?

---

So the reason for this is to render errors occurring in the Time Alignment pop-up next to their respective interfaces. If uncaught, you'd have to fail the entire request—while this allows us to limit failure to the impacted interface.

---

Whether and how to log these errors in the log file is a separate—and more interesting—question. Is that the desired outcome for you?

---

Hmmm.. I see

OK we'll tackle that in a follow-up. Would probably opt to dump the list of compiled errors all at once